tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -612,6 +682,49 @@ def file_lines(self) -> List[str]:
         )
         return file_text.splitlines()
 
+    def nth_instantiation(
+        self, fn: chapel.Function, idx: int
+    ) -> chapel.TypedSignature:
+        """
+        Given a function, return its nth instantiation. This uses the list of
+        instantiations collected while rebuilding the index.
+        """
+        return next(
+            itertools.islice(self.instantiations[fn.unique_id()], idx, None)
+        )
+
+    def instantiation_index(
+        self, fn: chapel.Function, sig: chapel.TypedSignature
+    ) -> int:
+        """
+        Given a function and an instantiation of that function, returns the
+        instantiation's index in the list of all instantiations.
+        """
```

Comment:
these functions are symmetric of each other? Thoughts on naming them in a way that implies that? `instantiation_at_index` and `index_of_instantiation`?